support-lib/jni/djinni_support.cpp

```diff
@@ -22,6 +22,7 @@
 #include <cstring>
 
 static_assert(sizeof(jlong) >= sizeof(void*), "must be able to fit a void* into a jlong");
+static_assert(sizeof(wchar_t) == 2 || sizeof(wchar_t) == 4, "wchar_t must be represented by UTF-16 or UTF-32 encoding");
```

Comment:
If the expression inside static_assert has no dependent names it is evaluated even if not instantiated.